nix/lib/default.nix

```diff
@@ -579,17 +580,48 @@ in rec {
       ]
     , dockerEnv ? []
     , dockerContents ? []
+    , dockerUser ? "app"
+    , dockerUserId ? "1001"
+    , dockerGroup ? "app"
+    , dockerGroupId ? "1001"
     , passthru ? {}
     , inTesting ? true
     , inStaging ? true
     , inProduction ? false
     , gunicornWorkers ? 3
+    , githubCommit  ? "unknown"
+    , taskGroupId  ? null
     }:
     let
+      version_json = {
+        inherit version;
+        source = "https://github.com/mozilla-releng/services";
+        commit = githubCommit;
+        build =
+          if taskGroupId != null
+            then "https://tools.taskcluster.net/groups/${taskGroupId}"
+            else "unknown";
+      };
       self = mkPython (args // {
 
         buildInputs = [ releng_pkgs.postgresql.package ] ++ buildInputs;
 
+        # Setup Dockerflow per https://github.com/mozilla-services/Dockerflow/
+        runAsRoot = if dockerUser == ""
+          then null
+          else ''
+            #!${stdenv.shell}
+            ${dockerTools.shadowSetup}
+            groupadd --gid ${dockerGroupId} ${dockerGroup}
+            useradd --gid ${dockerGroup} --uid ${dockerUserId} ${dockerUser}
+            # gunicorn requires /tmp, /var/tmp, or /usr/tmp
+            mkdir -p --mode=1777 /tmp
+            mkdir -p /app
```

Comment:
I would copy source code of our app as dockerflow instructions tells us, even if it is not being used. eg. ``cp -R ${self.src}/* /app``